gtest/src/wasm_executor.rs

```diff
@@ -0,0 +1,311 @@
+use codec::Encode;
+use core_processor::{Ext, ProcessorExt};
+use gear_backend_wasmtime::{env::StoreData, funcs_tree};
+use gear_core::{
+    env::{Ext as ExtTrait, ExtCarrier},
+    gas::{GasAllowanceCounter, GasCounter, ValueCounter},
+    memory::{AllocationsContext, PageBuf, PageNumber, WasmPageNumber},
+    message::{IncomingMessage, MessageContext},
+    program::Program,
+};
+use std::{collections::BTreeMap, mem};
+use wasmtime::{
+    Config, Engine, Extern, Func, Instance, Memory as WasmtimeMemory, MemoryType, Module, Store,
+    Val,
+};
+
+/// Binary meta-functions executor for testing purposes
+pub struct WasmExecutor {
+    instance: Instance,
+    store: Store<StoreData<Ext>>,
+    memory: WasmtimeMemory,
+}
+
+impl WasmExecutor {
+    /// Creates a WasmExecutor instance from a program.
+    /// Also uses provided memory pages for future execution
+    pub fn new(
+        program: &Program,
+        memory_pages: &BTreeMap<PageNumber, Box<PageBuf>>,
+        message: Option<IncomingMessage>,
```

Comment:
this message is excess, it's just the executor with preparation

Reply:
in general as I understood you wanted to change message argument to payload (no problem with that) and to move it from executor constructor to execution method. I am really confused about the second part. As I see the building levels in constructor: payload -> message -> messageContext -> ext -> store -> executor
Payload is injected on the first step and  couldn't be changed later so for me there is no good way to relocate message (payload) from constructor to execution method

Reply:
This method is for only one use, no need to change payload in current impl